<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Stephen Pederson" />

<meta name="date" content="2023-07-06" />

<title>ZR75: ERa</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">compare_diffsignal</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    ZR-75-1
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="zr75_ar.html">AR</a>
    </li>
    <li>
      <a href="zr75_er.html">ER</a>
    </li>
    <li>
      <a href="zr75_h3k27ac.html">H3K27ac</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">ZR75: ERa</h1>
<h4 class="author">Stephen Pederson</h4>
<h4 class="date">2023-07-06</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-07-07
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>compare_diffsignal/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230706code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230706)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230706code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230706)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsmpedcomparediffsignaltree873b7c71a1e65d8805280a3a47d321da62149159targetblank873b7c7a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/smped/compare_diffsignal/tree/873b7c71a1e65d8805280a3a47d321da62149159" target="_blank">873b7c7</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsmpedcomparediffsignaltree873b7c71a1e65d8805280a3a47d321da62149159targetblank873b7c7a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/smped/compare_diffsignal/tree/873b7c71a1e65d8805280a3a47d321da62149159" target="_blank">873b7c7</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    data/bam/

Untracked files:
    Untracked:  data/macs2/
    Untracked:  output/PRJNA509779/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/zr75_er.Rmd</code>) and HTML
(<code>docs/zr75_er.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/smped/compare_diffsignal/blob/873b7c71a1e65d8805280a3a47d321da62149159/analysis/zr75_er.Rmd" target="_blank">873b7c7</a>
</td>
<td>
Stephen Pederson
</td>
<td>
2023-07-07
</td>
<td>
Added sliding QLF methods
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/smped/compare_diffsignal/873b7c71a1e65d8805280a3a47d321da62149159/docs/zr75_er.html" target="_blank">873b7c7</a>
</td>
<td>
Stephen Pederson
</td>
<td>
2023-07-07
</td>
<td>
Added sliding QLF methods
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(extraChIPs)
library(Rsamtools)
library(rtracklayer)
library(scales)
library(glue)
library(plyranges)
library(pander)
library(DiffBind)
library(edgeR)
library(csaw)
library(BSgenome.Hsapiens.UCSC.hg19)
library(qsmooth)
library(BiocParallel)
register(MulticoreParam(threads))
theme_set(
  theme_bw() + 
    theme(
      text = element_text(size = 14),
      plot.title = element_text(hjust = 0.5),
      plot.subtitle = element_text(face = &quot;italic&quot;, hjust = 0.5)
    )
)
status_colours &lt;- c(
  Increased = &quot;red2&quot;, Decreased = &quot;deepskyblue3&quot;, 
  Unchanged = &quot;grey30&quot;, Undetected = &quot;grey70&quot;
)</code></pre>
<pre class="r"><code>samples &lt;- here::here(&quot;data&quot;, glue(&quot;{dataset}.tsv&quot;)) %&gt;% 
  read_tsv() %&gt;% 
  dplyr::filter(target == .GlobalEnv$target) %&gt;% 
  mutate(treatment = factor(treatment, levels = treat_levels))</code></pre>
<pre class="r"><code>bfl &lt;- samples %&gt;% 
  mutate(path = here::here(&quot;data&quot;, &quot;bam&quot;, dataset, glue(&quot;{accession}.sorted.bam&quot;))) %&gt;% 
  pull(path) %&gt;% 
  BamFileList()
sq &lt;- seqinfo(bfl) %&gt;% 
  keepStandardChromosomes() %&gt;% 
  dropSeqlevels(value = &quot;chrM&quot;) %&gt;% 
  sortSeqlevels() 
genome(sq) &lt;- &quot;GRCh37&quot;
isCircular(sq) &lt;- rep_len(FALSE, length(seqlevels(sq)))</code></pre>
<div id="peaks" class="section level2">
<h2>Peaks</h2>
<pre class="r"><code>greylist &lt;- here::here(&quot;data&quot;, glue(&quot;{samples$input}_greylist.bed&quot;)) %&gt;% 
  unique() %&gt;% 
  import.bed(which = GRanges(sq)) %&gt;% 
  keepStandardChromosomes()%&gt;% 
  dropSeqlevels(value = &quot;chrM&quot;) %&gt;% 
  sortSeqlevels() 
seqinfo(greylist) &lt;- sq
all_peaks &lt;- samples %&gt;% 
  mutate(peak_file = here::here(&quot;data&quot;, &quot;macs2&quot;, dataset, accession, glue(&quot;{accession}_peaks.narrowPeak&quot;))) %&gt;% 
  split(.$treatment) %&gt;% 
  lapply(pull, peak_file) %&gt;% 
  lapply(importPeaks, centre = TRUE, seqinfo = sq, glueNames = &quot;{str_remove_all(basename(x), &#39;_peaks.+&#39;)}&quot;)
treat_peaks &lt;- all_peaks %&gt;% 
  lapply(makeConsensus, p = 2/3, var = &quot;centre&quot;) %&gt;% 
  lapply(mutate, centre = vapply(centre, median, numeric(1))) 
merged_peaks &lt;- here::here(
  &quot;data&quot;, &quot;macs2&quot;, dataset, target, glue(&quot;{treat_levels}_merged_peaks.narrowPeak&quot;)
) %&gt;% 
  importPeaks(
    seqinfo = sq, centre = TRUE, glueNames = &quot;{str_remove_all(basename(x), &#39;_merged.+&#39;)}&quot;
  )
merged_peaks &lt;- treat_levels %&gt;% 
  lapply(
    function(x) filter_by_overlaps(merged_peaks[[x]], treat_peaks[[x]])
  ) %&gt;% 
  GRangesList() %&gt;% 
  setNames(treat_levels)
union_peaks &lt;- makeConsensus(merged_peaks, var = c(&quot;score&quot;, &quot;centre&quot;)) %&gt;% 
  mutate(
    centre = round(vapply(centre, mean, numeric(1)), 0),
    score=  vapply(score, max, numeric(1))
  ) %&gt;% 
  filter_by_non_overlaps(greylist)
union_peaks %&gt;% 
  select(score) %&gt;% 
  export.bed(file.path(out_path, glue(&quot;{target}_union_peaks.bed&quot;)))
centred_peaks &lt;- union_peaks %&gt;% 
  mutate(rng = paste0(seqnames, &quot;:&quot;, centre)) %&gt;% 
  colToRanges(&quot;rng&quot;) %&gt;% 
  resize(width = params$width, fix = &#39;center&#39;)
centred_peaks %&gt;% 
  select(score) %&gt;% 
  export.bed(file.path(out_path, glue(&quot;{target}_centred_peaks.bed&quot;)))</code></pre>
<p>Peaks were:</p>
<ol style="list-style-type: decimal">
<li>Imported for each sample individually</li>
<li>Merged to create treatment-specific ranges with peaks in 2 or more
samples</li>
<li>Those derived by merging samples (as input to
<code>macs2 callpeak</code>) were filtered by overlaps with the above
merged samples</li>
<li>A set of union peaks were then taken as the union of the above
treatment-specific peaks which passed filtering</li>
</ol>
<p>This gave 7,402 union peaks derived from <em>7,117</em> and
<em>6,887</em> treatment-specific peaks from E2 and E2DHT
respectively.</p>
</div>
<div id="fixed-width-analysis" class="section level2">
<h2>Fixed-Width Analysis</h2>
<div id="diffbind" class="section level3 tabset">
<h3 class="tabset">DiffBind</h3>
<pre class="r"><code>db_sample_sheet &lt;- samples %&gt;%
  mutate(
    Tissue = cell_line, Factor = target,# Condition = treat_levels[[1]], # Leads to two identically named masks
    bamReads = path(bfl),
    ControlID = &quot;Pooled&quot;,
    bamControl = file.path(dirname(bamReads), glue(&quot;{input}.sorted.bam&quot;)),
    Peaks = file.path(out_path, glue(&quot;{target}_centred_peaks.bed&quot;)),
    PeakCaller = &quot;bed&quot;
  ) %&gt;% 
  dplyr::rename(SampleID = accession, Treatment = treatment) %&gt;% 
  mutate(Replicate = seq_along(Treatment), .by = Treatment) %&gt;% 
  dplyr::select(
    SampleID, Tissue, Factor, #Condition, 
    Treatment, Replicate,
    bamReads, ControlID, bamControl, starts_with(&quot;Peak&quot;)
  ) %&gt;% 
  as.data.frame()
dba &lt;- dba(
  sampleSheet = db_sample_sheet, 
  config = data.frame(fragmentSize = params$fl, cores = threads)
) %&gt;% 
  dba.count(summits = floor(params$width / 2)) %&gt;% 
  dba.blacklist(cores = threads)</code></pre>
<p>For the initial analysis using DiffBind, all files were passed to the
requisite functions using a fixed-width of 400 bp, assuming a fragment
length of 195.</p>
<div id="library-size" class="section level4">
<h4>Library Size</h4>
<pre class="r"><code>dba_ls &lt;- dba.normalize(dba, method = DBA_ALL_METHODS, normalize = DBA_NORM_LIB) %&gt;% 
  dba.analyze(method = DBA_ALL_METHODS)
db_ls_results &lt;-  dba_ls %&gt;% 
  dba.report(method = DBA_EDGER, th = 1) %&gt;% 
  sortSeqlevels() %&gt;% 
  sort() 
db_ls_results$status &lt;- addDiffStatus(db_ls_results, fc_col = &quot;Fold&quot;)$status
seqinfo(db_ls_results) &lt;- sq</code></pre>
<p>This analysis returned 188 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>p &lt;- db_ls_results %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(Conc, Fold)) +
  geom_point(aes(colour = status), alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: DiffBind&quot;), subtitle = &quot;Library Size Normalisation&quot;) +
  labs(x = &quot;Ave Signal&quot;, y = &quot;logFC&quot;, colour = &quot;Status&quot;) +
  scale_colour_manual(values = status_colours)
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
<div class="figure">
<img src="assets/PRJNA509779/ERa/plotma-db-ls.png" alt="" />
<p class="caption"><em>MA-plot for DiffBind results using Library Size
Normalisation</em></p>
</div>
</div>
<div id="tmm" class="section level4">
<h4>TMM</h4>
<pre class="r"><code>dba_tmm &lt;- dba.normalize(dba, method = DBA_ALL_METHODS, normalize = DBA_NORM_TMM) %&gt;% 
  dba.analyze(method = DBA_ALL_METHODS)
db_tmm_results &lt;- dba_tmm %&gt;%
  dba.report(method = DBA_EDGER, th = 1) %&gt;%
  sortSeqlevels() %&gt;%
  sort()
db_tmm_results$status &lt;- addDiffStatus(db_tmm_results, fc_col = &quot;Fold&quot;)$status
seqinfo(db_tmm_results) &lt;- sq</code></pre>
<p>This analysis returned 590 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>p &lt;- db_tmm_results  %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(Conc, Fold)) +
  geom_point(aes(colour = status), alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: DiffBind&quot;), subtitle = &quot;TMM Normalisation&quot;) +
  labs(x = &quot;Ave Signal&quot;, y = &quot;logFC&quot;, colour = &quot;Status&quot;) +
  scale_colour_manual(values = status_colours)
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
<div class="figure">
<img src="assets/PRJNA509779/ERa/plotma-db-tmm.png" alt="" />
<p class="caption"><em>MA-plot for DiffBind results using TMM
Normalisation</em></p>
</div>
</div>
</div>
<div id="extrachips" class="section level3 tabset">
<h3 class="tabset">extraChIPs</h3>
<pre class="r"><code>se &lt;- regionCounts(bfl, centred_peaks, ext = 200)
colData(se) %&lt;&gt;%
  as_tibble(rownames = &quot;sample&quot;) %&gt;% 
  mutate(
    accession = str_remove_all(sample, &quot;.sorted.bam&quot;),
  ) %&gt;% 
  left_join(samples, by = &quot;accession&quot;) %&gt;% 
  dplyr::select(all_of(colnames(colData(se))), sample, accession, treatment) %&gt;% 
  as.data.frame() %&gt;% 
  column_to_rownames(&quot;sample&quot;) %&gt;% 
  DataFrame()
X &lt;- model.matrix(~treatment, data = colData(se))</code></pre>
<div id="library-size-1" class="section level4">
<h4>Library Size</h4>
<pre class="r"><code>ec_ls_results &lt;- fitAssayDiff(se, design = X, asRanges = TRUE) %&gt;% 
  addDiffStatus()</code></pre>
<p>This analysis returned 181 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>p &lt;- ec_ls_results %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(logCPM, logFC)) +
  geom_point(aes(colour = status), alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: extraChIPs&quot;), subtitle = &quot;Library Size Normalisation&quot;) +
  labs(colour = &quot;Status&quot;) +
  scale_colour_manual(values = status_colours)
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
<div class="figure">
<img src="assets/PRJNA509779/ERa/plotma-ec-ls-fixed.png" alt="" />
<p class="caption"><em>MA-plot for extraChIPs results using Library Size
Normalisation</em></p>
</div>
</div>
<div id="tmm-1" class="section level4">
<h4>TMM</h4>
<pre class="r"><code>ec_tmm_results &lt;- fitAssayDiff(se, design = X, norm = &quot;TMM&quot;, asRanges = TRUE) %&gt;% 
  addDiffStatus()</code></pre>
<p>This analysis returned 600 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>p &lt;- ec_tmm_results %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(logCPM, logFC)) +
  geom_point(aes(colour = status), alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: extraChIPs&quot;), subtitle = &quot;TMM Normalisation&quot;) +
  labs(colour = &quot;Status&quot;) +
  scale_colour_manual(values = status_colours)
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
<div class="figure">
<img src="assets/PRJNA509779/ERa/plotma-ec-tmm-fixed.png" alt="" />
<p class="caption"><em>MA-plot for extraChIPs results using TMM
Normalisation</em></p>
</div>
</div>
<div id="tmm-groups" class="section level4">
<h4>TMM Groups</h4>
<pre class="r"><code>ec_tmmg_results &lt;- fitAssayDiff(se, design = X, norm = &quot;TMM&quot;, groups = &quot;treat&quot;, asRanges = TRUE) %&gt;% 
  addDiffStatus()</code></pre>
<p>This analysis returned 208 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>p &lt;- ec_tmmg_results %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(logCPM, logFC)) +
  geom_point(aes(colour = status), alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: extraChIPs&quot;), subtitle = &quot;TMM Normalisation (Groups)&quot;) +
  labs(colour = &quot;Status&quot;) +
  scale_colour_manual(values = status_colours)
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
<div class="figure">
<img src="assets/PRJNA509779/ERa/plotma-ec-tmmg-fixed.png" alt="" />
<p class="caption"><em>MA-plot for extraChIPs results using TMM-groups
Normalisation</em></p>
</div>
</div>
</div>
<div id="smooth-quantile" class="section level3 tabset">
<h3 class="tabset">Smooth Quantile</h3>
<pre class="r"><code>bs &lt;- BSgenome.Hsapiens.UCSC.hg19
seqlevels(centred_peaks) &lt;- seqlevels(bs)
genome(centred_peaks) &lt;- &quot;hg19&quot;
seqinfo(centred_peaks) &lt;- seqinfo(bs)
gc &lt;- getSeq(bs, centred_peaks) %&gt;% 
  letterFrequency(letters = &quot;GC&quot;, as.prob = TRUE)
rowData(se)$gc &lt;- gc[,1]</code></pre>
<div id="sqn" class="section level4">
<h4>SQN</h4>
<pre class="r"><code>assay(se, &quot;normCounts&quot;) &lt;- qsmooth(se, group_factor = se$treatment) %&gt;% 
  qsmoothData()
sqn_results &lt;- fitAssayDiff(
  se, assay = &quot;normCounts&quot;, lib.size = NULL, design = X, asRanges = TRUE
) %&gt;% 
  addDiffStatus()</code></pre>
<p>This analysis returned 605 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>p &lt;- sqn_results %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(logCPM, logFC)) +
  geom_point(aes(colour = status), alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: extraChIPs&quot;), subtitle = &quot;SQ Normalisation&quot;) +
  labs(colour = &quot;Status&quot;) +
  scale_colour_manual(values = status_colours)
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
<div class="figure">
<img src="assets/PRJNA509779/ERa/plotma-ec-sqn-fixed.png" alt="" />
<p class="caption"><em>MA-plot for extraChIPs results using Smooth
Quantile Normalisation</em></p>
</div>
</div>
<div id="sqn-gc" class="section level4">
<h4>SQN-GC</h4>
<pre class="r"><code>assay(se, &quot;normCountsGC&quot;) &lt;- qsmoothGC(
  se, group_factor = se$treatment, gc = rowData(se)$gc
) %&gt;% 
  qsmoothData()
sqngc_results &lt;- fitAssayDiff(
  se, assay = &quot;normCountsGC&quot;, lib.size = NULL, design = X, asRanges = TRUE
) %&gt;% 
  addDiffStatus()</code></pre>
<p>This analysis returned 277 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>p &lt;- sqngc_results %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(logCPM, logFC)) +
  geom_point(aes(colour = status), alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: extraChIPs&quot;), subtitle = &quot;SQ-GC Normalisation&quot;) +
  labs(colour = &quot;Status&quot;) +
  scale_colour_manual(values = status_colours)
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
<div class="figure">
<img src="assets/PRJNA509779/ERa/plotma-ec-sqngc-fixed.png" alt="" />
<p class="caption"><em>MA-plot for extraChIPs results using Smooth
Quantile Normalisation, accounting for GC content</em></p>
</div>
</div>
</div>
<div id="comparison-of-extrachips-and-diffbind" class="section level3">
<h3>Comparison of extraChIPs and DiffBind</h3>
<p>Given the expectation of identical results, those from DiffBind and
extraChIPs were compared for both Library-Size and TMM
normalisation.</p>
<pre class="r"><code>dir_combs &lt;- c(&quot;DiffBind&quot;, &quot;extraChIPs&quot;) %&gt;% 
vapply(
  function(x) paste(x, setdiff(names(status_colours), &quot;Undetected&quot;)),
  character(3)
) %&gt;% 
  as.data.frame() %&gt;% 
  expand.grid() %&gt;% 
  mutate(
    status = paste(extraChIPs, DiffBind, sep = &quot; - &quot;),
    status = case_when(
      grepl(&quot;Increased.+Increased&quot;, status) ~ &quot;Both Increased&quot;,
      grepl(&quot;Decreased.+Decreased&quot;, status) ~ &quot;Both Decreased&quot;,
      grepl(&quot;Unchanged.+Unchanged&quot;, status) ~ &quot;Both Unchanged&quot;,
      TRUE ~ status
    )
  ) %&gt;% 
  pull(status) %&gt;% 
  sort()
dir_cols &lt;- c(
  rgb(0, 0, 1), rgb(1, 0, 0), rgb(0.5, 0.5, 0.5), #Both
  &quot;orange&quot;, rgb(0, 0.5, 1), &quot;orange&quot;, rgb(1, 0.5, 0),
  rgb(0.5, 0, 1), rgb(1, 0, 0.5)
)
names(dir_cols) &lt;- dir_combs
stat_cols &lt;- c(
  Neither = &quot;grey70&quot;, Both = &quot;forestgreen&quot;,
  `DiffBind Only` = &quot;red&quot;, `extraChIPs Only` = &quot;orange&quot;
)</code></pre>
<div id="library-size-normalisation" class="section level4 tabset">
<h4 class="tabset">Library Size Normalisation</h4>
<div id="logfc-estimates" class="section level5">
<h5>logFC Estimates</h5>
<pre class="r"><code>p &lt;- GRangesList(
  DiffBind = select(db_ls_results, logFC = Fold, status), 
  extraChIPs = select(ec_ls_results, logFC, status)
) %&gt;% 
  plotPairwise(var = &quot;logFC&quot;, colour = &quot;status&quot;, xside = &quot;none&quot;, yside = &quot;none&quot;) +
  ggtitle(
    glue(&quot;{target} logFC&quot;), &quot;Library-Size Normalisation&quot;
  ) +
  scale_colour_manual(values = dir_cols)
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
<div class="figure">
<img src="assets/PRJNA509779/ERa/compare-logfc-db-ec-ls.png" alt="" />
<p class="caption"><em>Comparison of logFC estimates returned using
library-size normalisation for both DiffBind and extraChIPs</em></p>
</div>
</div>
<div id="p-values" class="section level5">
<h5>P-Values</h5>
<pre class="r"><code>p &lt;- GRangesList(
  DiffBind = select(db_ls_results, PValue = `p-value`, status), 
  extraChIPs = select(ec_ls_results, PValue, status)
) %&gt;% 
  mapGrlCols(var = c(&quot;PValue&quot;, &quot;status&quot;)) %&gt;% # , collapse = &quot;status&quot;) %&gt;% 
  as_tibble() %&gt;% 
  mutate(
    across(ends_with(&quot;PValue&quot;), function(x) -log10(x)),
    status = case_when(
      grepl(&quot;(In|De)creased&quot;, DiffBind_status) &amp; grepl(&quot;(In|De)creased&quot;, extraChIPs_status) ~ &quot;Both&quot;, 
      !grepl(&quot;(In|De)creased&quot;, DiffBind_status) &amp; grepl(&quot;(In|De)creased&quot;, extraChIPs_status) ~ &quot;extraChIPs Only&quot;, 
      grepl(&quot;(In|De)creased&quot;, DiffBind_status) &amp; !grepl(&quot;(In|De)creased&quot;, extraChIPs_status) ~ &quot;DiffBind Only&quot;, 
      TRUE ~ &quot;Neither&quot;
    )
  ) %&gt;% 
  ggplot(aes(DiffBind_PValue, extraChIPs_PValue)) +
  geom_point(aes(colour = status)) +
  ggtitle(
    glue(&quot;{target}&quot;), &quot;Library-Size Normalisation&quot;
  ) +
  scale_colour_manual(values = stat_cols) +
  labs(
    x = expr(paste(-log[10], &quot;p (DiffBind)&quot;)),
    y = expr(paste(-log[10], &quot;p (extraChIPs)&quot;)),
    colour = &quot;Status&quot;
  )
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
<pre><code>Warning: Removed 6 rows containing missing values (`geom_point()`).</code></pre>
<div class="figure">
<img src="assets/PRJNA509779/ERa/compare-p-db-ec-ls.png" alt="" />
<p class="caption"><em>Comparison of -log<sub>10</sub> p-values returned
using library-size normalisation for both DiffBind and
extraChIPs</em></p>
</div>
</div>
</div>
<div id="tmm-normalisation" class="section level4 tabset">
<h4 class="tabset">TMM Normalisation</h4>
<div id="logfc-estimates-1" class="section level5">
<h5>logFC Estimates</h5>
<pre class="r"><code>p &lt;- GRangesList(
  DiffBind = select(db_tmm_results, logFC = Fold, status), 
  extraChIPs = select(ec_tmm_results, logFC, status)
) %&gt;% 
  plotPairwise(var = &quot;logFC&quot;, colour = &quot;status&quot;, xside = &quot;none&quot;, yside = &quot;none&quot;) +
  ggtitle(
    glue(&quot;{target} logFC&quot;), &quot;TMM Normalisation&quot;
  ) +
  scale_colour_manual(values = dir_cols)
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
<div class="figure">
<img src="assets/PRJNA509779/ERa/compare-logfc-db-ec-tmm.png" alt="" />
<p class="caption"><em>Comparison of logFC estimates returned using TMM
normalisation for both DiffBind and extraChIPs</em></p>
</div>
</div>
<div id="p-values-1" class="section level5">
<h5>P-Values</h5>
<pre class="r"><code>p &lt;- GRangesList(
  DiffBind = select(db_tmm_results, PValue = `p-value`, status), 
  extraChIPs = select(ec_tmm_results, PValue, status)
) %&gt;% 
  mapGrlCols(var = c(&quot;PValue&quot;, &quot;status&quot;)) %&gt;% # , collapse = &quot;status&quot;) %&gt;% 
  as_tibble() %&gt;% 
  mutate(
    across(ends_with(&quot;PValue&quot;), function(x) -log10(x)),
    status = case_when(
      grepl(&quot;(In|De)creased&quot;, DiffBind_status) &amp; grepl(&quot;(In|De)creased&quot;, extraChIPs_status) ~ &quot;Both&quot;, 
      !grepl(&quot;(In|De)creased&quot;, DiffBind_status) &amp; grepl(&quot;(In|De)creased&quot;, extraChIPs_status) ~ &quot;extraChIPs Only&quot;, 
      grepl(&quot;(In|De)creased&quot;, DiffBind_status) &amp; !grepl(&quot;(In|De)creased&quot;, extraChIPs_status) ~ &quot;DiffBind Only&quot;, 
      TRUE ~ &quot;Neither&quot;
    )
  ) %&gt;% 
  ggplot(aes(DiffBind_PValue, extraChIPs_PValue)) +
  geom_point(aes(colour = status)) +
  ggtitle(
    glue(&quot;{target}&quot;), &quot;TMM Normalisation&quot;
  ) +
  scale_colour_manual(values = stat_cols) +
  labs(
    x = expr(paste(-log[10], &quot;p (DiffBind)&quot;)),
    y = expr(paste(-log[10], &quot;p (extraChIPs)&quot;)),
    colour = &quot;Status&quot;
  )
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
<pre><code>Warning: Removed 6 rows containing missing values (`geom_point()`).</code></pre>
<div class="figure">
<img src="assets/PRJNA509779/ERa/compare-p-db-ec-tmm.png" alt="" />
<p class="caption"><em>Comparison of -log<sub>10</sub> p-values returned
using TMM normalisation for both DiffBind and extraChIPs</em></p>
</div>
</div>
</div>
</div>
</div>
<div id="sliding-windows" class="section level2">
<h2>Sliding Windows</h2>
<pre class="r"><code>data(&quot;hg19.blacklist&quot;, package = &quot;GreyListChIP&quot;) 
window_size &lt;- round(params$width / 4, -1)
window_step &lt;- ceiling(0.1 * window_size / 3) * 10
rp &lt;- readParam(
  pe = &quot;none&quot;, dedup = TRUE,
  restrict = seqnames(sq),
  discard = c(greylist, hg19.blacklist)
)
bfl &lt;- path(bfl) %&gt;% 
  dirname() %&gt;% 
  unique() %&gt;% 
  file.path(glue(&quot;{samples$input}.sorted.bam&quot;)) %&gt;% 
  unique() %&gt;% 
  BamFileList() %&gt;% 
  setNames(unique(samples$input)) %&gt;% 
  c(bfl)</code></pre>
<p>Windows were set to be about 1/4 of the fixed-width window-size, with
each range covered up to 3 times. This gave 100bp windows moving in
increments of 40bp.</p>
<pre class="r"><code>wincounts &lt;- windowCounts(
  bam.files = bfl,
  spacing = window_step,
  width = window_size,
  ext = params$fl,
  filter = length(bfl),
  param = rp
)
seqlevels(wincounts) &lt;- seqlevels(sq)
seqinfo(wincounts) &lt;- sq
colData(wincounts) %&lt;&gt;%
  as_tibble(rownames = &quot;sample&quot;) %&gt;% 
  mutate(
    accession = str_remove_all(sample, &quot;.sorted.bam&quot;),
    ChIP_Target = ifelse(accession %in% samples$accession, .GlobalEnv$target, &quot;Input&quot;)
  ) %&gt;% 
  left_join(samples, by = &quot;accession&quot;) %&gt;% 
  dplyr::select(all_of(colnames(colData(se))), sample, accession, ChIP_Target, treatment) %&gt;% 
  as.data.frame() %&gt;% 
  column_to_rownames(&quot;sample&quot;) %&gt;% 
  DataFrame()</code></pre>
<pre class="r"><code>filtcounts &lt;- dualFilter(
  x = wincounts[, wincounts$ChIP_Target == target],
  bg = wincounts[, wincounts$ChIP_Target != target], 
  ref = union_peaks,
  q = 0.6
)</code></pre>
<pre class="r"><code>filtRanges &lt;- granges(rowRanges(filtcounts))
seqlevels(filtRanges) &lt;- seqlevels(bs)
genome(filtRanges) &lt;- &quot;hg19&quot;
seqinfo(filtRanges) &lt;- seqinfo(bs)
gc &lt;- getSeq(bs, filtRanges) %&gt;% 
  letterFrequency(letters = &quot;GC&quot;, as.prob = TRUE)
rowRanges(filtcounts)$gc &lt;- gc[,1]
assay(filtcounts, &quot;sq_counts&quot;) &lt;- assay(filtcounts, &quot;counts&quot;) %&gt;% 
  qsmooth(group_factor = filtcounts$treatment) %&gt;% 
  qsmoothData()
assay(filtcounts, &quot;sq_logCPM&quot;) &lt;- assay(filtcounts, &quot;logCPM&quot;) %&gt;% 
  qsmooth(group_factor = filtcounts$treatment) %&gt;% 
  qsmoothData()
assay(filtcounts, &quot;sqgc_counts&quot;) &lt;- assay(filtcounts, &quot;counts&quot;) %&gt;% 
  qsmoothGC(filtcounts$treatment, rowRanges(filtcounts)$gc) %&gt;% 
  qsmoothData()
assay(filtcounts, &quot;sqgc_logCPM&quot;) &lt;- assay(filtcounts, &quot;logCPM&quot;) %&gt;% 
  qsmoothGC(filtcounts$treatment, rowRanges(filtcounts)$gc) %&gt;% 
  qsmoothData()
X &lt;- model.matrix(~treatment, data = colData(filtcounts))</code></pre>
<div id="library-size-2" class="section level3">
<h3>Library Size</h3>
<pre class="r"><code>sw_ls_all &lt;- fitAssayDiff(filtcounts, design = X, asRanges = TRUE) 
sw_ls_results &lt;- sw_ls_all %&gt;% 
  mergeByHMP(inc_cols = &quot;overlaps_ref&quot;) %&gt;% 
  filter(n_windows &gt; 1) %&gt;% 
  mutate(FDR = p.adjust(hmp, &quot;fdr&quot;)) %&gt;% 
  select(overlaps_ref, keyval_range, starts_with(&quot;log&quot;), PValue = hmp, FDR) %&gt;% 
  addDiffStatus()</code></pre>
<p>This analysis returned 107 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>p &lt;- sw_ls_all %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(logCPM, logFC)) +
  geom_point(alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: Sliding Windows&quot;), subtitle = &quot;Library Size Normalisation&quot;) +
  labs(x = &quot;Ave Signal&quot;, y = &quot;logFC&quot;) 
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
<div class="figure">
<img src="assets/PRJNA509779/ERa/plotma-sw-ls.png" alt="" />
<p class="caption"><em>MA-plot for sliding-window results using Library
Size Normalisation</em></p>
</div>
</div>
<div id="tmm-2" class="section level3">
<h3>TMM</h3>
<pre class="r"><code>sw_tmm_all &lt;- fitAssayDiff(filtcounts, design = X, norm = &quot;TMM&quot;, asRanges = TRUE) 
sw_tmm_results &lt;- sw_tmm_all %&gt;% 
  mergeByHMP(inc_cols = &quot;overlaps_ref&quot;) %&gt;% 
  filter(n_windows &gt; 1) %&gt;% 
  mutate(FDR = p.adjust(hmp, &quot;fdr&quot;)) %&gt;% 
  select(overlaps_ref, keyval_range, starts_with(&quot;log&quot;), PValue = hmp, FDR) %&gt;% 
  addDiffStatus()</code></pre>
<p>This analysis returned 370 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>p &lt;- sw_tmm_results %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(logCPM, logFC)) +
  geom_point(alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: Sliding Windows&quot;), subtitle = &quot;TMM Normalisation&quot;) +
  labs(x = &quot;Ave Signal&quot;, y = &quot;logFC&quot;) 
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
<div class="figure">
<img src="assets/PRJNA509779/ERa/plotma-sw-tmm.png" alt="" />
<p class="caption"><em>MA-plot for sliding-window results using TMM
Normalisation</em></p>
</div>
</div>
<div id="tmm-groups-1" class="section level3">
<h3>TMM Groups</h3>
<pre class="r"><code>sw_tmmg_all &lt;- fitAssayDiff(
  filtcounts, design = X, norm = &quot;TMM&quot;, groups = &quot;treatment&quot;, asRanges = TRUE
) 
sw_tmmg_results &lt;- sw_tmmg_all %&gt;% 
  mergeByHMP(inc_cols = &quot;overlaps_ref&quot;) %&gt;% 
  filter(n_windows &gt; 1) %&gt;% 
  mutate(FDR = p.adjust(hmp, &quot;fdr&quot;)) %&gt;% 
  select(overlaps_ref, keyval_range, starts_with(&quot;log&quot;), PValue = hmp, FDR) %&gt;% 
  addDiffStatus()</code></pre>
<p>This analysis returned 129 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>p &lt;- sw_tmmg_results %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(logCPM, logFC)) +
  geom_point(alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: Sliding Windows&quot;), subtitle = &quot;TMM Normalisation (Groups)&quot;) +
  labs(x = &quot;Ave Signal&quot;, y = &quot;logFC&quot;) 
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
<div class="figure">
<img src="assets/PRJNA509779/ERa/plotma-sw-tmmg.png" alt="" />
<p class="caption"><em>MA-plot for sliding-window results using TMM
Normalisation within treatment groups</em></p>
</div>
</div>
<div id="smooth-quantile-1" class="section level3">
<h3>Smooth Quantile</h3>
<pre class="r"><code>sw_sq_all &lt;- fitAssayDiff(
  filtcounts, assay = &quot;sq_counts&quot;, totals = NULL, design = X, asRanges = TRUE
) 
sw_sq_results &lt;- sw_sq_all %&gt;% 
  mergeByHMP(inc_cols = &quot;overlaps_ref&quot;) %&gt;% 
  filter(n_windows &gt; 1) %&gt;% 
  mutate(FDR = p.adjust(hmp, &quot;fdr&quot;)) %&gt;% 
  select(overlaps_ref, keyval_range, starts_with(&quot;log&quot;), PValue = hmp, FDR) %&gt;% 
  addDiffStatus()</code></pre>
<p>This analysis returned 320 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>p &lt;- sw_sq_all %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(logCPM, logFC)) +
  geom_point(alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: Sliding Windows&quot;), subtitle = &quot;SQ Normalisation&quot;) +
  labs(x = &quot;Ave Signal&quot;, y = &quot;logFC&quot;) 
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
</div>
<div id="smooth-quantile-gc" class="section level3">
<h3>Smooth Quantile GC</h3>
<pre class="r"><code>sw_sqgc_all &lt;- fitAssayDiff(
  filtcounts, assay = &quot;sqgc_counts&quot;, totals = NULL, design = X, asRanges = TRUE
) 
sw_sqgc_results &lt;- sw_sqgc_all %&gt;% 
  mergeByHMP(inc_cols = &quot;overlaps_ref&quot;) %&gt;% 
  filter(n_windows &gt; 1) %&gt;% 
  mutate(FDR = p.adjust(hmp, &quot;fdr&quot;)) %&gt;% 
  select(overlaps_ref, keyval_range, starts_with(&quot;log&quot;), PValue = hmp, FDR) %&gt;% 
  addDiffStatus()</code></pre>
<p>This analysis returned 342 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>p &lt;- sw_sqgc_all %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(logCPM, logFC)) +
  geom_point(alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: Sliding Windows&quot;), subtitle = &quot;SQ-GC Normalisation&quot;) +
  labs(x = &quot;Ave Signal&quot;, y = &quot;logFC&quot;) 
if (interactive()) {
  p
} else {
  lb &lt;- knitr::opts_current$get(&quot;label&quot;)
  fig_out &lt;- file.path(fig_path, glue(lb, &quot;.png&quot;))
  href &lt;- str_remove_all(fig_out, &quot;.+docs/&quot;)
  png(
    fig_out, res = 300, units = &quot;in&quot;,
    height = knitr::opts_chunk$get(&quot;fig.height&quot;), 
    width = knitr::opts_chunk$get(&quot;fig.width&quot;)
  )
  print(p)
  dev.off()
}</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
