<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Stephen Pederson" />

<meta name="date" content="2023-07-06" />

<title>ZR75: AR</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">compare_diffsignal</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    ZR-75-1
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="zr75_ar.html">AR</a>
    </li>
    <li>
      <a href="zr75_er.html">ER</a>
    </li>
    <li>
      <a href="zr75_h3k27ac.html">H3K27ac</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">ZR75: AR</h1>
<h4 class="author">Stephen Pederson</h4>
<h4 class="date">2023-07-06</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-07-07
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>compare_diffsignal/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongobjectspresent">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Environment:</strong> objects present
</a>
</p>
</div>
<div id="strongEnvironmentstrongobjectspresent"
class="panel-collapse collapse">
<div class="panel-body">
<p>
The global environment had objects present when the code in the R
Markdown file was run. These objects can affect the analysis in your R
Markdown file in unknown ways. For reproduciblity it’s best to always
run the code in an empty environment. Use <code>wflow_publish</code> or
<code>wflow_build</code> to ensure that the code is always run in an
empty environment.
</p>
<p>
The following objects were defined in the global environment when these
results were created:
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
Name
</th>
<th style="text-align:left;">
Class
</th>
<th style="text-align:left;">
Size
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
all_peaks
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
8.5 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
bfl
</td>
<td style="text-align:left;">
BamFileList
</td>
<td style="text-align:left;">
5.9 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
bs
</td>
<td style="text-align:left;">
BSgenome
</td>
<td style="text-align:left;">
86.7 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
cell_line
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
112 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
centred_peaks
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
639.7 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
dataset
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
120 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
db_ls_deseq
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
1.7 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
db_ls_deseq2
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
1.8 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
db_ls_edger
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
1.8 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
db_ls_results
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
1.7 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
db_sample_sheet
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
4.8 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
db_tmm_deseq
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
1.7 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
db_tmm_deseq2
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
1.8 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
db_tmm_edger
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
1.8 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
db_tmm_results
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
1.7 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
dba
</td>
<td style="text-align:left;">
DBA
</td>
<td style="text-align:left;">
7.8 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
dba_ls
</td>
<td style="text-align:left;">
DBA
</td>
<td style="text-align:left;">
28.7 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
dba_tmm
</td>
<td style="text-align:left;">
DBA
</td>
<td style="text-align:left;">
28.7 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
dge
</td>
<td style="text-align:left;">
DGEList
</td>
<td style="text-align:left;">
2.8 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
ec_ls_results
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
1.2 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
ec_tmm_results
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
1.2 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
ec_tmmg_results
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
1.2 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
gc
</td>
<td style="text-align:left;">
matrix;array
</td>
<td style="text-align:left;">
115.8 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
greylist
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
43.7 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
ls_res
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
995.1 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
merged_peaks
</td>
<td style="text-align:left;">
CompressedGRangesList
</td>
<td style="text-align:left;">
2.3 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
out_path
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
232 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
samples
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
3.5 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
se
</td>
<td style="text-align:left;">
RangedSummarizedExperiment
</td>
<td style="text-align:left;">
2.4 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
sq
</td>
<td style="text-align:left;">
Seqinfo
</td>
<td style="text-align:left;">
3.1 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
sqn_results
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
1.3 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
sqngc_results
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
1.3 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
status_colours
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
760 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
target
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
112 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
threads
</td>
<td style="text-align:left;">
numeric
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
tmm_res
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
995.1 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
treat_levels
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
176 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
treat_peaks
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
684.5 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
union_peaks
</td>
<td style="text-align:left;">
GRanges
</td>
<td style="text-align:left;">
590 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
X
</td>
<td style="text-align:left;">
matrix;array
</td>
<td style="text-align:left;">
1.9 Kb
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230706code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230706)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230706code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230706)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsmpedcomparediffsignaltree6a9ecfbbda6c66e8223b1ab947b1c322bdbe00b1targetblank6a9ecfba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/smped/compare_diffsignal/tree/6a9ecfbbda6c66e8223b1ab947b1c322bdbe00b1" target="_blank">6a9ecfb</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsmpedcomparediffsignaltree6a9ecfbbda6c66e8223b1ab947b1c322bdbe00b1targetblank6a9ecfba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/smped/compare_diffsignal/tree/6a9ecfbbda6c66e8223b1ab947b1c322bdbe00b1" target="_blank">6a9ecfb</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    data/bam/

Untracked files:
    Untracked:  data/PRJNA509779.tsv
    Untracked:  data/SRR8315192_greylist.bed
    Untracked:  data/macs2/
    Untracked:  output/PRJNA509779/

Unstaged changes:
    Modified:   README.md
    Modified:   analysis/zr75_ar.Rmd
    Modified:   code/_method_comparison.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/zr75_ar.Rmd</code>) and HTML
(<code>docs/zr75_ar.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/smped/compare_diffsignal/blob/6a9ecfbbda6c66e8223b1ab947b1c322bdbe00b1/analysis/zr75_ar.Rmd" target="_blank">6a9ecfb</a>
</td>
<td>
Stephen Pederson
</td>
<td>
2023-07-06
</td>
<td>
Initial Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/smped/compare_diffsignal/6a9ecfbbda6c66e8223b1ab947b1c322bdbe00b1/docs/zr75_ar.html" target="_blank">6a9ecfb</a>
</td>
<td>
Stephen Pederson
</td>
<td>
2023-07-06
</td>
<td>
Initial Commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(extraChIPs)
library(Rsamtools)
library(rtracklayer)
library(scales)
library(glue)
library(plyranges)
library(pander)
library(DiffBind)
library(edgeR)
library(csaw)
library(BSgenome.Hsapiens.UCSC.hg19)
library(qsmooth)
theme_set(
  theme_bw() + 
    theme(
      text = element_text(size = 14),
      plot.title = element_text(hjust = 0.5),
      plot.subtitle = element_text(face = &quot;italic&quot;, hjust = 0.5)
    )
)
status_colours &lt;- c(Increased = &quot;red2&quot;, Decreased = &quot;deepskyblue3&quot;, Unchanged = &quot;grey30&quot;, Undetected = &quot;grey70&quot;)</code></pre>
<pre class="r"><code>samples &lt;- here::here(&quot;data&quot;, glue(&quot;{dataset}.tsv&quot;)) %&gt;% 
  read_tsv() %&gt;% 
  dplyr::filter(target == .GlobalEnv$target) %&gt;% 
  mutate(treatment = factor(treatment, levels = treat_levels))</code></pre>
<pre class="r"><code>bfl &lt;- samples %&gt;% 
  mutate(path = here::here(&quot;data&quot;, &quot;bam&quot;, dataset, glue(&quot;{accession}.sorted.bam&quot;))) %&gt;% 
  pull(path) %&gt;% 
  BamFileList()
sq &lt;- seqinfo(bfl) %&gt;% 
  keepStandardChromosomes() %&gt;% 
  dropSeqlevels(value = &quot;chrM&quot;) %&gt;% 
  sortSeqlevels() 
genome(sq) &lt;- &quot;GRCh37&quot;
isCircular(sq) &lt;- rep_len(FALSE, length(seqlevels(sq)))</code></pre>
<div id="peaks" class="section level2">
<h2>Peaks</h2>
<pre class="r"><code>greylist &lt;- here::here(&quot;data&quot;, glue(&quot;{samples$input}_greylist.bed&quot;)) %&gt;% 
  unique() %&gt;% 
  import.bed(which = GRanges(sq)) %&gt;% 
  keepStandardChromosomes()%&gt;% 
  dropSeqlevels(value = &quot;chrM&quot;) %&gt;% 
  sortSeqlevels() 
seqinfo(greylist) &lt;- sq
all_peaks &lt;- samples %&gt;% 
  mutate(peak_file = here::here(&quot;data&quot;, &quot;macs2&quot;, dataset, accession, glue(&quot;{accession}_peaks.narrowPeak&quot;))) %&gt;% 
  split(.$treatment) %&gt;% 
  lapply(pull, peak_file) %&gt;% 
  lapply(importPeaks, centre = TRUE, seqinfo = sq, glueNames = &quot;{str_remove_all(basename(x), &#39;_peaks.+&#39;)}&quot;)
treat_peaks &lt;- all_peaks %&gt;% 
  lapply(makeConsensus, p = 2/3, var = &quot;centre&quot;) %&gt;% 
  lapply(mutate, centre = vapply(centre, median, numeric(1))) 
merged_peaks &lt;- here::here(
  &quot;data&quot;, &quot;macs2&quot;, dataset, target, glue(&quot;{treat_levels}_merged_peaks.narrowPeak&quot;)
) %&gt;% 
  importPeaks(
    seqinfo = sq, centre = TRUE, glueNames = &quot;{str_remove_all(basename(x), &#39;_merged.+&#39;)}&quot;
  )
merged_peaks &lt;- treat_levels %&gt;% 
  lapply(
    function(x) filter_by_overlaps(merged_peaks[[x]], treat_peaks[[x]])
  ) %&gt;% 
  GRangesList() %&gt;% 
  setNames(treat_levels)
union_peaks &lt;- makeConsensus(merged_peaks, var = c(&quot;score&quot;, &quot;centre&quot;)) %&gt;% 
  mutate(
    centre = round(vapply(centre, mean, numeric(1)), 0),
    score=  vapply(score, max, numeric(1))
  ) %&gt;% 
  filter_by_non_overlaps(greylist)
union_peaks %&gt;% 
  select(score) %&gt;% 
  export.bed(file.path(out_path, glue(&quot;{target}_union_peaks.bed&quot;)))
centred_peaks &lt;- union_peaks %&gt;% 
  mutate(rng = paste0(seqnames, &quot;:&quot;, centre)) %&gt;% 
  colToRanges(&quot;rng&quot;) %&gt;% 
  resize(width = params$width, fix = &#39;center&#39;)
centred_peaks %&gt;% 
  select(score) %&gt;% 
  export.bed(file.path(out_path, glue(&quot;{target}_centred_peaks.bed&quot;)))</code></pre>
<p>Peaks were:</p>
<ol style="list-style-type: decimal">
<li>Imported for each sample individually</li>
<li>Merged to create treatment-specific ranges with peaks in 2 or more
samples</li>
<li>Those derived by merging samples (as input to
<code>macs2 callpeak</code>) were filtered by overlaps with the above
merged samples</li>
<li>A set of union peaks were then taken as the union of the above
treatment-specific peaks which passed filtering</li>
</ol>
<p>This gave 14,763 union peaks derived from <em>1,017</em> and
<em>15,874</em> treatment-specific peaks from E2 and E2DHT
respectively.</p>
</div>
<div id="fixed-width-analysis" class="section level2">
<h2>Fixed-Width Analysis</h2>
<div id="diffbind" class="section level3 tabset">
<h3 class="tabset">DiffBind</h3>
<pre class="r"><code>db_sample_sheet &lt;- samples %&gt;%
  mutate(
    Tissue = cell_line, Factor = target,# Condition = treat_levels[[1]], # Leads to two identically named masks
    bamReads = path(bfl),
    ControlID = &quot;Pooled&quot;,
    bamControl = file.path(dirname(bamReads), glue(&quot;{input}.sorted.bam&quot;)),
    Peaks = file.path(out_path, glue(&quot;{target}_centred_peaks.bed&quot;)),
    PeakCaller = &quot;bed&quot;
  ) %&gt;% 
  dplyr::rename(SampleID = accession, Treatment = treatment) %&gt;% 
  mutate(Replicate = seq_along(Treatment), .by = Treatment) %&gt;% 
  dplyr::select(
    SampleID, Tissue, Factor, #Condition, 
    Treatment, Replicate,
    bamReads, ControlID, bamControl, starts_with(&quot;Peak&quot;)
  ) %&gt;% 
  as.data.frame()
dba &lt;- dba(
  sampleSheet = db_sample_sheet, 
  config = data.frame(fragmentSize = params$fl, cores = threads)
) %&gt;% 
  dba.count(summits = floor(params$width / 2)) %&gt;% 
  dba.blacklist(cores = threads)</code></pre>
<pre><code>coverage: 13012992 bp (0.43%)</code></pre>
<p>For the initial analysis using DiffBind, all files were passed to the
requisite functions using a fixed-width of 400 bp, assuming a fragment
length of 195.</p>
<p><strong>DiffBind Results as shown currently use DESeq2 due to a bug
in DiffBind 3.10</strong></p>
<div id="library-size" class="section level4">
<h4>Library Size</h4>
<pre class="r"><code>dba_ls &lt;- dba.normalize(dba, normalize = DBA_NORM_LIB) %&gt;% 
  dba.analyze(method = DBA_ALL_METHODS)
# db_ls_results &lt;-  dba_ls %&gt;% 
#   dba.report(method = DBA_EDGER, th = 1) %&gt;% 
#   sortSeqlevels() %&gt;% 
#   sort()
# seqinfo(db_ls_results) &lt;- sq
## DiffBind currently has a bug
db_ls_results &lt;-  dba_ls %&gt;% 
  dba.report(method = DBA_DESEQ2, th = 1) %&gt;% 
  sortSeqlevels() %&gt;% 
  sort()
seqinfo(db_ls_results) &lt;- sq</code></pre>
<p>This analysis returned 12,249 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>db_ls_results %&gt;%
  addDiffStatus(fc_col = &quot;Fold&quot;) %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(Conc, Fold)) +
  geom_point(aes(colour = status), alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: DiffBind&quot;), subtitle = &quot;Library Size Normalisation&quot;) +
  labs(x = &quot;Ave Signal&quot;, y = &quot;logFC&quot;, colour = &quot;Status&quot;) +
  scale_colour_manual(values = status_colours)</code></pre>
<img src="figure/_method_comparison.Rmd/plotma-db-ls-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
</div>
<div id="tmm" class="section level4">
<h4>TMM</h4>
<pre class="r"><code>dba_tmm &lt;- dba.normalize(dba, normalize = DBA_NORM_TMM) %&gt;% 
  dba.analyze(method = DBA_ALL_METHODS)
# db_tmm_results &lt;- dba_tmm %&gt;% 
#   dba.report(method = DBA_EDGER, th = 1) %&gt;% 
#   sortSeqlevels() %&gt;% 
#   sort()
# seqinfo(db_tmm_results) &lt;- sq
db_tmm_results &lt;- dba_tmm %&gt;% 
  dba.report(method = DBA_DESEQ2, th = 1) %&gt;% 
  sortSeqlevels() %&gt;% 
  sort()
seqinfo(db_tmm_results) &lt;- sq</code></pre>
<p>This analysis returned 4,221 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>db_tmm_results %&gt;% 
  addDiffStatus(fc_col = &quot;Fold&quot;) %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(Conc, Fold)) +
  geom_point(aes(colour = status), alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: DiffBind&quot;), subtitle = &quot;TMM Normalisation&quot;) +
  labs(x = &quot;Ave Signal&quot;, y = &quot;logFC&quot;, colour = &quot;Status&quot;) +
  scale_colour_manual(values = status_colours)</code></pre>
<img src="figure/_method_comparison.Rmd/plotma-db-tmm-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
</div>
</div>
<div id="extrachips" class="section level3 tabset">
<h3 class="tabset">extraChIPs</h3>
<pre class="r"><code>se &lt;- regionCounts(bfl, centred_peaks, ext = 200)
colData(se) %&lt;&gt;%
  as_tibble(rownames = &quot;sample&quot;) %&gt;% 
  mutate(
    accession = str_remove_all(sample, &quot;.sorted.bam&quot;),
  ) %&gt;% 
  left_join(samples, by = &quot;accession&quot;) %&gt;% 
  dplyr::select(all_of(colnames(colData(se))), sample, accession, treatment) %&gt;% 
  as.data.frame() %&gt;% 
  column_to_rownames(&quot;sample&quot;) %&gt;% 
  DataFrame()
X &lt;- model.matrix(~treatment, data = colData(se))</code></pre>
<div id="library-size-1" class="section level4">
<h4>Library Size</h4>
<pre class="r"><code>ec_ls_results &lt;- fitAssayDiff(se, design = X, asRanges = TRUE) %&gt;% 
  addDiffStatus()</code></pre>
<p>This analysis returned 12,769 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>ec_ls_results %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(logCPM, logFC)) +
  geom_point(aes(colour = status), alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: extraChIPs&quot;), subtitle = &quot;Library Size Normalisation&quot;) +
  labs(colour = &quot;Status&quot;) +
  scale_colour_manual(values = status_colours)</code></pre>
<img src="figure/_method_comparison.Rmd/plotma-ec-ls-fixed-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
</div>
<div id="tmm-1" class="section level4">
<h4>TMM</h4>
<pre class="r"><code>ec_tmm_results &lt;- fitAssayDiff(se, design = X, norm = &quot;TMM&quot;, asRanges = TRUE) %&gt;% 
  addDiffStatus()</code></pre>
<p>This analysis returned 5,738 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>ec_tmm_results %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(logCPM, logFC)) +
  geom_point(aes(colour = status), alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: extraChIPs&quot;), subtitle = &quot;TMM Normalisation&quot;) +
  labs(colour = &quot;Status&quot;) +
  scale_colour_manual(values = status_colours)</code></pre>
<img src="figure/_method_comparison.Rmd/plotma-ec-tmm-fixed-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
</div>
<div id="tmm-groups" class="section level4">
<h4>TMM Groups</h4>
<pre class="r"><code>ec_tmmg_results &lt;- fitAssayDiff(se, design = X, norm = &quot;TMM&quot;, groups = &quot;treat&quot;, asRanges = TRUE) %&gt;% 
  addDiffStatus()</code></pre>
<p>This analysis returned 12,703 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>ec_tmmg_results %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(logCPM, logFC)) +
  geom_point(aes(colour = status), alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: extraChIPs&quot;), subtitle = &quot;TMM Normalisation (Groups)&quot;) +
  labs(colour = &quot;Status&quot;) +
  scale_colour_manual(values = status_colours)</code></pre>
<img src="figure/_method_comparison.Rmd/plotma-ec-tmmg-fixed-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
</div>
</div>
<div id="smooth-quantile" class="section level3 tabset">
<h3 class="tabset">Smooth Quantile</h3>
<pre class="r"><code>bs &lt;- BSgenome.Hsapiens.UCSC.hg19
seqlevels(centred_peaks) &lt;- seqlevels(bs)
genome(centred_peaks) &lt;- &quot;hg19&quot;
seqinfo(centred_peaks) &lt;- seqinfo(bs)
gc &lt;- getSeq(bs, centred_peaks) %&gt;% 
  letterFrequency(letters = &quot;GC&quot;, as.prob = TRUE)
rowData(se)$gc &lt;- gc[,1]</code></pre>
<div id="sqn" class="section level4">
<h4>SQN</h4>
<pre class="r"><code>assay(se, &quot;normCounts&quot;) &lt;- qsmooth(se, group_factor = se$treatment) %&gt;% 
  qsmoothData()
sqn_results &lt;- fitAssayDiff(
  se, assay = &quot;normCounts&quot;, lib.size = NULL, design = X, asRanges = TRUE
) %&gt;% 
  addDiffStatus()</code></pre>
<p>This analysis returned 5,656 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>sqn_results %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(logCPM, logFC)) +
  geom_point(aes(colour = status), alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: extraChIPs&quot;), subtitle = &quot;SQ Normalisation&quot;) +
  labs(colour = &quot;Status&quot;) +
  scale_colour_manual(values = status_colours)</code></pre>
<img src="figure/_method_comparison.Rmd/plotma-ec-sqn-fixed-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
</div>
<div id="sqn-gc" class="section level4">
<h4>SQN-GC</h4>
<pre class="r"><code>assay(se, &quot;normCountsGC&quot;) &lt;- qsmoothGC(
  se, group_factor = se$treatment, gc = rowData(se)$gc
) %&gt;% 
  qsmoothData()
sqngc_results &lt;- fitAssayDiff(
  se, assay = &quot;normCountsGC&quot;, lib.size = NULL, design = X, asRanges = TRUE
) %&gt;% 
  addDiffStatus()</code></pre>
<p>This analysis returned 5,463 peaks considered to show evidence of
differential signal.</p>
<pre class="r"><code>sqngc_results %&gt;% 
  as_tibble() %&gt;% 
  ggplot(aes(logCPM, logFC)) +
  geom_point(aes(colour = status), alpha = 0.7) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 0) +
  ggtitle(glue(&quot;{target}: extraChIPs&quot;), subtitle = &quot;SQ-GC Normalisation&quot;) +
  labs(colour = &quot;Status&quot;) +
  scale_colour_manual(values = status_colours)</code></pre>
<img src="figure/_method_comparison.Rmd/plotma-ec-sqngc-fixed-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /data/tki_bodl/sw/R/4.3.0-gcc-blas-fullsupport/lib64/R/lib/libRblas.so 
LAPACK: /data/tki_bodl/sw/R/4.3.0-gcc-blas-fullsupport/lib64/R/lib/libRlapack.so;  LAPACK version 3.11.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C               LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8    LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Australia/Perth
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] qsmooth_1.16.0                    BSgenome.Hsapiens.UCSC.hg19_1.4.3 BSgenome_1.68.0                  
 [4] csaw_1.34.0                       edgeR_3.42.4                      limma_3.56.2                     
 [7] DiffBind_3.10.0                   pander_0.6.5                      plyranges_1.20.0                 
[10] glue_1.6.2                        scales_1.2.1                      rtracklayer_1.60.0               
[13] Rsamtools_2.16.0                  Biostrings_2.68.1                 XVector_0.40.0                   
[16] extraChIPs_1.5.6                  SummarizedExperiment_1.30.2       Biobase_2.60.0                   
[19] MatrixGenerics_1.12.2             matrixStats_1.0.0                 ggside_0.2.2                     
[22] GenomicRanges_1.52.0              GenomeInfoDb_1.36.0               IRanges_2.34.0                   
[25] S4Vectors_0.38.1                  BiocGenerics_0.46.0               BiocParallel_1.34.2              
[28] lubridate_1.9.2                   forcats_1.0.0                     stringr_1.5.0                    
[31] dplyr_1.1.2                       purrr_1.0.1                       readr_2.1.4                      
[34] tidyr_1.3.0                       tibble_3.2.1                      ggplot2_3.4.2                    
[37] tidyverse_2.0.0                   workflowr_1.7.0                  

loaded via a namespace (and not attached):
  [1] fs_1.6.2                   ProtGenerics_1.32.0        bitops_1.0-7               httr_1.4.6                
  [5] RColorBrewer_1.1-3         doParallel_1.0.17          InteractionSet_1.28.1      numDeriv_2016.8-1.1       
  [9] tools_4.3.0                backports_1.4.1            utf8_1.2.3                 R6_2.5.1                  
 [13] mgcv_1.8-42                lazyeval_0.2.2             Gviz_1.44.0                apeglm_1.22.1             
 [17] GetoptLong_1.0.5           withr_2.5.0                prettyunits_1.1.1          gridExtra_2.3             
 [21] VennDiagram_1.7.3          cli_3.6.1                  formatR_1.14               sass_0.4.6                
 [25] labeling_0.4.2             SQUAREM_2021.1             mvtnorm_1.2-2              genefilter_1.82.1         
 [29] mixsqp_0.3-48              foreign_0.8-84             dichromat_2.0-0.1          invgamma_1.1              
 [33] bbmle_1.0.25               rstudioapi_0.14            RSQLite_2.3.1              generics_0.1.3            
 [37] shape_1.4.6                BiocIO_1.10.0              hwriter_1.3.2.1            gtools_3.9.4              
 [41] vroom_1.6.3                Matrix_1.5-4               interp_1.1-4               futile.logger_1.4.3       
 [45] fansi_1.0.4                lifecycle_1.0.3            whisker_0.4.1              yaml_2.3.7                
 [49] gplots_3.1.3               BiocFileCache_2.8.0        grid_4.3.0                 blob_1.2.4                
 [53] promises_1.2.0.1           crayon_1.5.2               bdsmatrix_1.3-6            lattice_0.21-8            
 [57] ComplexUpset_1.3.3         annotate_1.78.0            GenomicFeatures_1.52.0     KEGGREST_1.40.0           
 [61] pillar_1.9.0               knitr_1.43                 ComplexHeatmap_2.16.0      metapod_1.8.0             
 [65] rjson_0.2.21               systemPipeR_2.6.3          codetools_0.2-19           ShortRead_1.58.0          
 [69] getPass_0.2-2              GreyListChIP_1.32.0        data.table_1.14.8          vctrs_0.6.3               
 [73] png_0.1-8                  gtable_0.3.3               amap_0.8-19                emdbook_1.3.13            
 [77] cachem_1.0.8               xfun_0.39                  S4Arrays_1.0.4             coda_0.19-4               
 [81] survival_3.5-5             iterators_1.0.14           GenomicInteractions_1.34.0 nlme_3.1-162              
 [85] bit64_4.0.5                progress_1.2.2             filelock_1.0.2             rprojroot_2.0.3           
 [89] bslib_0.5.0                irlba_2.3.5.1              KernSmooth_2.23-20         rpart_4.1.19              
 [93] colorspace_2.1-0           DBI_1.1.3                  Hmisc_5.1-0                nnet_7.3-18               
 [97] DESeq2_1.40.2              tidyselect_1.2.0           processx_3.8.1             bit_4.0.5                 
[101] compiler_4.3.0             curl_5.0.1                 git2r_0.32.0               htmlTable_2.4.1           
[105] xml2_1.3.4                 DelayedArray_0.26.3        checkmate_2.2.0            caTools_1.18.2            
[109] callr_3.7.3                rappdirs_0.3.3             digest_0.6.31              rmarkdown_2.22            
[113] htmltools_0.5.5            pkgconfig_2.0.3            jpeg_0.1-10                base64enc_0.1-3           
[117] highr_0.10                 dbplyr_2.3.2               fastmap_1.1.1              ensembldb_2.24.0          
[121] rlang_1.1.1                GlobalOptions_0.1.2        htmlwidgets_1.6.2          jquerylib_0.1.4           
[125] EnrichedHeatmap_1.30.0     farver_2.1.1               jsonlite_1.8.5             VariantAnnotation_1.46.0  
[129] RCurl_1.98-1.12            magrittr_2.0.3             Formula_1.2-5              GenomeInfoDbData_1.2.10   
[133] patchwork_1.1.2            munsell_0.5.0              Rcpp_1.0.10                stringi_1.7.12            
[137] zlibbioc_1.46.0            MASS_7.3-58.4              plyr_1.8.8                 parallel_4.3.0            
[141] ggrepel_0.9.3              deldir_1.0-9               splines_4.3.0              hms_1.1.3                 
[145] circlize_0.4.15            locfit_1.5-9.8             ps_1.7.5                   igraph_1.5.0              
[149] biomaRt_2.56.0             futile.options_1.0.1       XML_3.99-0.14              evaluate_0.21             
[153] latticeExtra_0.6-30        biovizBase_1.48.0          lambda.r_1.2.4             BiocManager_1.30.21       
[157] tzdb_0.4.0                 foreach_1.5.2              tweenr_2.0.2               httpuv_1.6.11             
[161] polyclip_1.10-4            clue_0.3-64                ashr_2.2-54                ggforce_0.4.1             
[165] xtable_1.8-4               broom_1.0.5                restfulr_0.0.15            AnnotationFilter_1.24.0   
[169] later_1.3.1                truncnorm_1.0-9            memoise_2.0.1              AnnotationDbi_1.62.1      
[173] GenomicAlignments_1.36.0   cluster_2.1.4              sva_3.48.0                 timechange_0.2.0          
[177] here_1.0.1                </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
